.github/workflows/sonarcloud.yaml

```diff
@@ -0,0 +1,51 @@
+name: Sonarcloud Scan
```

Comment:
can you link some documentation regarding how this job works? there should be some context to follow; providing a README would be best, but reference links would also be sufficient for someone new to this code to understand what its purpose and how it works

Reply:
see https://docs.sonarcloud.io/getting-started/github/

Reply:
please add it as a comment inside the file so it can be easily found. also it would be good to describe the relationship between this job and the resulting badge in this PR : https://github.com/Alluxio/alluxio/pull/17003#issuecomment-1453297889

Reply:
You also need to manually create a project, refer to https://bestpractices.coreinfrastructure.org/en/projects

Reply:
Manually create an Alluxio organization to report results, please see https://sonarcloud.io/explore/projects?sort=-analysis_date

Reply:
there is a lot of information you are sharing that the maintainers do not know about... all these details you are providing in the PR will not be recorded into the repo unless you commit them

Reply:
You can refer to the projects that have integrated Scorecards in the community to learn